### DISCUSSION TITLE
mariaDbDriver.execute is not using backticks as quotes

### DISCUSSION THREAD

**Discussion by maciossek** [mikro-orm/mikro-orm#6185] [General]
### Describe the bug

When executing `mariaDbDriver.execute("DROP DATABASE IF EXISTS ?", [dbName])` I receive the following error

```
DROP DATABASE IF EXISTS 'test-db-22181' - (conn:1828, no: 1064, SQLState: 42000) You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ''test-db-22181'' at line 1
```

### Reproduction

I am using this vitest teardown function to clean up databases

```typescript
import { MikroORM } from "@mikro-orm/core";
import { MikroORM as MariaDbMikroOrm } from "@mikro-orm/mariadb";

import mariaDbConfig from "./src/database/mariadb-v1/mikro-orm.mariadb.config";
import postgreSqlconfig from "./src/database/postgresql-default/mikro-orm.postgres.config";

export async function teardown() {
  const postgreSqlOrm = await MikroORM.init(postgreSqlconfig);
  const postgreSqlDriver = postgreSqlOrm.em.getDriver();
  const postgreSqlDatabases = await postgreSqlDriver.execute(
    "SELECT datname FROM pg_database WHERE datistemplate = false;",
  );

  const mariaDbOrm = await MariaDbMikroOrm.init(mariaDbConfig);
  const mariaDbDriver = mariaDbOrm.em.getDriver();
  const mariaDbDatabases = await mariaDbDriver.execute(
    "SELECT SCHEMA_NAME AS datname FROM information_schema.SCHEMATA WHERE SCHEMA_NAME NOT IN ('information_schema', 'mysql', 'performance_schema', 'sys');",
  );

  const postgreSqlTestDatabases = postgreSqlDatabases
    .map((db: { datname: string }) => db.datname)
    .filter((dbName: string) => dbName.startsWith("test-db"));

  const mariaDbTestDatabases = mariaDbDatabases
    .map((db: { datname: string }) => db.datname)
    .filter((dbName: string) => dbName.startsWith("test-db"));

  const promises = [];

  for (const dbName of postgreSqlTestDatabases) {
    promises.push(postgreSqlDriver.execute("DROP DATABASE IF EXISTS ?", [dbName]));
  }

  for (const dbName of mariaDbTestDatabases) {
    promises.push(mariaDbDriver.execute("DROP DATABASE IF EXISTS ?", [dbName]));
  }

  await Promise.all(promises);
  await Promise.all([postgreSqlOrm.close(true), mariaDbOrm.close(true)]);
}

```

### What driver are you using?

@mikro-orm/mariadb

### MikroORM version

6.3.13

### Node.js version

22

### Operating system

Ubuntu

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
First of all, there is no reason to use a raw query here, just do this instead:

```ts
await orm.schema.dropDatabase();
```

Second, you shouldn't be using the driver method at all, either use `orm.schema.execute()` or `em.execute()`.

Lastly, your problem is using `?` which represents a value binding, and that is supposed to be quoted just like you see it, with single quotes. You need to use `??` to quote an identifier, which will use backticks.

**Reply by maciossek** [reply]
many thanks!
